ts/cart/cart.entity.ts

```diff
@@ -0,0 +1,51 @@
+import { CartItemType } from './cart.interface.js';
+
+export class CartItem implements CartItemType {
+  id: number;
+  name: string;
+  price: number;
+  image: string;
+  discount: number;
+  quantity: number;
+
+  constructor(cartItem: CartItemType) {
+    const { id, name, price, image, discount, quantity } = cartItem;
+    this.id = id;
+    this.name = name;
+    this.price = price;
+    this.image = image;
+    this.discount = discount || 0;
+    this.quantity = quantity;
```

Comment:
tạo thêm key discountPrice hoặc finalPrice để tính toán giá đã giảm ở đây, k cần tạo thêm function